### ISSUE TITLE
Long options don't work

### ISSUE THREAD

**Issue by papadakospan** [cifsd-team/ksmbd-tools#195]
I tested this with `ksmbd.control`. `-s` works, but `--shutdown` gives me the usage message. It would seem that the way `getopt` is used, it only accepts short options

**Comment by alexmyczko** [comment]
I can confirm this to be the case for me as well with 3.4.2.

**Comment by namjaejeon** [comment]
Yep, right.
Enzo is trying to update the whole ksmbd-tools. He will fix it after that.
@ematsumiya Right ? and We need to add manpage also if you change single binary.

**Comment by ematsumiya** [comment]
That's right. A manpage and updated docs should also be included in the final version.
I'm on vacation now though, so please expect a ready-for-review version by the end of January.

**Comment by namjaejeon** [comment]
Cool! Have a good vacation:)

**Comment by namjaejeon** [comment]
@ematsumiya Hi Enzo, Is there any update for single binary work ?

**Comment by ematsumiya** [comment]
@namjaejeon I just sent my patch series to linux-cifs ML and you. It was based on ksmbd-tools-next branch.

manpage is still missing, but the README has been updated at least. I hope to have some time this week or next to work on the manpage.

@papadakospan @alexmyczko I'd appreciate if you could test my proposal, feedback is welcome. Source is at https://github.com/ematsumiya/ksmbd-tools/tree/ksmbdctl-v2-rebase

**Comment by papadakospan** [comment]
It seems to work nicely, good work. You've included some useful features

**Comment by namjaejeon** [comment]
@ematsumiya Thanks for your patches. I will check it.

**Comment by papadakospan** [comment]
I have gotten global share options to work on [ematsumiya](https://github.com/ematsumiya)'s v2-rebase tree.
Should I wait until it gets merged here to make a pull request?

**Comment by namjaejeon** [comment]
> It was based on ksmbd-tools-next branch.

@ematsumiya It would be good if you create the patch-set base on #master branch.

**Comment by papadakospan** [comment]
@ematsumiya Your changes are very nice, I hope you make a pull request or something towards this repo

**Comment by ematsumiya** [comment]
Oops. Got side tracked yet again... sorry about the delay.

I think it's still required to base it on master, right @namjaejeon? I'll try to do it by tomorrow.

**Comment by namjaejeon** [comment]
@ematsumiya okay:) Thanks for your work!

**Comment by ematsumiya** [comment]
Submitted PR #265, rebased on cifsd-team/ksmbd-tools master. Testing and feedback appreciated.

**Comment by namjaejeon** [comment]
@ematsumiya Let me check it! Thanks!
